src/Refactoring-UI/ReRenameMethodDriver.class.st

```diff
@@ -191,32 +191,29 @@ ReRenameMethodDriver >> requestNewMessage [
 ReRenameMethodDriver >> runRefactoring [
 
 	| failedConditions |
+		
 	self configureRefactoring.
-	[ newMessage := self requestNewMessage.
-	"note that it would be better to pass a method name to the refactoring instead of splitting the arguments this way"
-	refactoring newSelector: newMessage selector; permutation: newMessage permutation.
-	"If the name is the same then there is no need to perform the refactoring 
-	and in such case we can just escape."
-	refactoring newNameDoesNotRequireRefactoringPrecondition check
-		ifTrue: [ ^ self ].
-	shouldEscape ifTrue: [ ^ self ].
-	failedConditions := refactoring failedApplicabilityPreconditions.
-	failedConditions isEmpty ]
-		whileFalse: [
-			failedConditions do: [ :cond |
-				"We need to mockify it"
-				self informDialog
-					label: cond errorString;
-					title: 'Invalid method name';
-					openModal ] ].
+
+	failedConditions := OrderedCollection empty.	
+	[ 
+		newMessage := self requestNewMessage.
+		self configureMessage.
+		
+		refactoring isMethodDefinitionUnchanged ifTrue: [ ^ self ].
+		shouldEscape ifTrue: [ ^ self ]. "What is the use of this variable: shouldEscape?"
```

Comment:
User can invoke this refactoring by mistake or give up while typing new name (in `requestNewMessage`). If this happens, he clicks on `Cancel` and with this variable we propagate that cancel to escape this loop. Maybe there is a better way to do this?